### DISCUSSION TITLE
Trouble with "hold on" and plot vector3d data

### DISCUSSION THREAD

**Discussion by jeffrahl** [mtex-toolbox/mtex#1473] [Ask Anything]
I am trying to plot vector direction data points on the same figure, but the behavior of "hold on" is erratic. In some cases I am able to plot multiple points on the same plot, but in other cases some of the data do not appear. I constructed the simple script below to illustrate the problem - I have run this in both MTEX 5.8.1 and 5.7 on both a Mac and a PC and get the same results. Thanks for any insights you might have.

Jeff

---------------------

clear all
close all

% define three vectors for plotting
test1 = vector3d(0.080352, 0.504153, -0.859868);
test2 = vector3d(0.904446, -0.399462, -0.149692);
test3 = vector3d( 0.418953, 0.765676, 0.488076);

% all three points can be plotted individually...
figure
plot(test1)
figure
plot(test2)
figure
plot(test3)

% but plotting them all at once with "hold on" only shows some of them...
figure
plot(test1)
hold on
plot(test2)
plot(test3)

% ..though if ordered differently they sometimes do all appear (huh?!)
figure
plot(test3)
hold on
plot(test2)
plot(test1)

**Comment by zmichels** [comment]
I’ve never had this trouble. Is it possible the different hemispheres are an issue in your case?

If you add ‘antipodal’ (and maybe also specify a hemisphere like ‘lower’)… does that show them all?

Example:
plot(vec, ‘antipodal’, ‘lower’)

Zach

**Reply by jeffrahl** [reply]
Thanks, Zach, for the quick reply! This solution did indeed allow me to plot all of my data simultaneously - thanks! Still, I would like, if possible, to visualize the data using both hemispheres, in case anyone has an idea how to make that work. (Though I think I will be able to use this solution as a workaround.)

Thanks,

Jeff

**Comment by zmichels** [comment]
Hmmm… I wonder if you bundle them all together into a single variable and try plotting rather than use the “hold on” approach, which may default to the initial plot’s hemisphere.

Like:
Vecs = [v1 v2 v3]
plot(Vecs)

**Comment by jeffrahl** [comment]
I had that thought, but the trouble there is that I'd like to specify different symbols and colors to the individual points, and I'm not sure how to do that without overlaying plots.

It is strange - I was able to do this years ago with an earlier version of mtex, but I can't remember which version it was (and I also don't want to revert my code back).

Thanks much for all these ideas! I appreciate it.

Jeff

**Comment by zmichels** [comment]
You can use a vector of color triplets as another argument to the plot command… but you’d have to decide ahead of time what vectors to plot and what colors to assign them.

**Comment by zmichels** [comment]
I do recall a previous version where I could specify a vector for both size and color in a single plot command. You can still do color (and doesn’t have to be an RGB triplet I don’t think… just a numerical list/vector of same length)… but can no longer specify size that way, which is a bummer for some things I like plotting. Don’t know if a way to do it with symbols tho. Maybe a request to add that is worthwhile?

Z

**Comment by zmichels** [comment]
Ooops… wait… I think all you need in the initial plot is to specify ‘complete’

plot(vec, ‘antipodal’, ‘complete’)

Something like that

**Reply by jeffrahl** [reply]
YES!

Problem solved. Many thanks!

Jeff

**Comment by kilir** [comment]
Also, don't forget about 'add2all' - required in case you would not want to plot axes but actually vectors:
```
figure
plot(test1)
hold on
plot(test2,'add2all')
plot(test3,'add2all')
hold off
```

**Reply by jeffrahl** [reply]
Thanks, Rudiger! That is super-helpful!

Jeff
